app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventDetailRepositoryImpl.java

```diff
@@ -199,40 +199,36 @@ else if (d2.programModule().programStages().byProgramUid().eq(programUid).one().
     @Override
     public Single<Pair<CategoryCombo, List<CategoryOptionCombo>>> catOptionCombos() {
         return d2.programModule().programs().uid(programUid).get()
-                .filter(program -> program.categoryCombo() != null)
-                .flatMapSingle(program -> d2.categoryModule().categoryCombos().uid(program.categoryComboUid()).get())
-                .filter(categoryCombo -> !categoryCombo.isDefault())
-                .flatMapSingle(categoryCombo -> Single.zip(
-                        d2.categoryModule().categoryCombos()
-                                .uid(categoryCombo.uid()).get(),
-                        d2.categoryModule().categoryOptionCombos()
-                                .byCategoryComboUid().eq(categoryCombo.uid()).get(),
-                        Pair::create
-                ));
+                .map(program -> {
+                    CategoryCombo catCombo = d2.categoryModule().categoryCombos().uid(program.categoryComboUid()).blockingGet();
+                    List<CategoryOptionCombo> catOptCombos = d2.categoryModule().categoryOptionCombos()
+                            .byCategoryComboUid().eq(catCombo.uid()).blockingGet();
+                    return Pair.create(catCombo, catOptCombos);
+                });
```

Comment:
anything we should take into consideration for testing? :+1: 

pd: nice refactor